packages/ingest/S3ProviderClient.js

```diff
@@ -51,10 +51,10 @@ class S3ProviderClient {
     return objects.map(({ Key, Size, LastModified }) => ({
       name: basename(Key),
       // If the object is at the top level of the bucket, path.dirname is going
-      // to return "." as the dirname.  It should instead be null.
-      path: dirname(Key) === '.' ? null : dirname(Key),
+      // to return "." as the dirname.  It should instead be undefined.
+      path: dirname(Key) === '.' ? undefined : dirname(Key),
```

Comment:
This is a breaking change, and needs to be noted as such in the Changelog. (A good change, but still a breaking one.) If the calling code is checking for `null`, it will break.